name: meka_omnibase_control_mb2
version: iss
joint_array_component: m3joint_array_mb2
pwr_component: m3pwr_pwr029

param:
    xd_max: 0.25
    xdd_max: 0.25
    td_max: 0.50
    tdd_max: 0.50

    # pi/6, 2pi/3, -2pi/3, -pi/6 
    alpha: [1.0472, 2.0944, -2.0944, -1.0472] 
    l: [0.2678, 0.2678, 0.2678, 0.2678]
    d: [0.0200, 0.0200, 0.0200, 0.0200]
    r: [0.0550, 0.0550, 0.0550, 0.0550]
    beta_offset: [-0.2618, -0.2618, -0.2618, -0.2618] 
    beta_ratio: [1.0, 1.0, 1.0, 1.0] 
    phid_ratio: [-1.0, -1.0, -1.0, -1.0] 

    # TODO: Real values!
    betad_max: [1.0, 1.0, 1.0, 1.0]
    betadd_max: [1.0, 1.0, 1.0, 1.0]
    phid_max: [1.0, 1.0, 1.0, 1.0]
    phidd_max: [1.0, 1.0, 1.0, 1.0]

    casters:
        Ns: 3.0000 # 96/24
        Nt: 2.4286 # 85/35
        Nw: 2.0000 # 32/16
        k_p: -0.0003
        k_i: 0.0003
        k_d: 0.0000
        k_i_limit: 20.0
        k_i_range: 100.0

